packages/eslint/index.js

```diff
@@ -1,82 +1,81 @@
-const deepmerge = require('deepmerge');
-const clone = require('lodash.clonedeep');
-const omit = require('lodash.omit');
+const pick = require('lodash.pick');
+const { merge: eslintMerge } = require('eslint/lib/config/config-ops');
 const { ConfigurationError, DuplicateRuleError } = require('neutrino/errors');
 
-const merge = (source, destination) => {
```

Comment:
This old merge function missed several cases:
* it didn't apply the rule-merge to `baseConfig.rules`
* the rule-merge doesn't keep the settings of the old rule, when wanting to just adjust the off/warn/error level (unlike eslint's own merge)
* it doesn't merge `overrides` correctly

See these testcases:
https://github.com/eslint/eslint/blob/5984820670c3085becc98b75827fc4899001e605/tests/lib/config/config-ops.js#L305-L319
https://github.com/eslint/eslint/blob/5984820670c3085becc98b75827fc4899001e605/tests/lib/config/config-ops.js#L481-L494